src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java

```diff
@@ -270,45 +294,51 @@ private static Subject loginAsSubject()
         }
 
         @Override
-        public byte[] initialResponse()
+        public CompletionStage<ByteBuffer> initialResponse()
         {
 
             if (saslClient.hasInitialResponse())
             {
                 try
                 {
-                    return Subject.doAs(subject, (PrivilegedExceptionAction<byte[]>) () ->
+                    byte[] response = Subject.doAs(subject, (PrivilegedExceptionAction<byte[]>) () ->
                             saslClient.evaluateChallenge(new byte[0]));
+                    return CompletableFuture.completedFuture(ByteBuffer.wrap(response));
                 } catch (PrivilegedActionException e)
                 {
                     throw new RuntimeException(e.getException());
                 }
             }
 
-            return new byte[0];
+            return CompletableFuture.completedFuture(ByteBuffer.wrap(new byte[0]));
         }
 
         @Override
-        public byte[] evaluateChallenge(final byte[] challenge) {
-
+        public CompletionStage<ByteBuffer> evaluateChallenge(@Nullable ByteBuffer challenge) {
             try
             {
-                return Subject.doAs(subject, (PrivilegedExceptionAction<byte[]>) () ->
-                        saslClient.evaluateChallenge(challenge));
+                byte[] bytes = new byte[challenge.capacity()];
+                challenge.get(bytes, 0, bytes.length);
+
+                byte[] evaluation = Subject.doAs(subject, (PrivilegedExceptionAction<byte[]>) () ->
+                        saslClient.evaluateChallenge(bytes));
+                return CompletableFuture.completedFuture(ByteBuffer.wrap(evaluation));
             } catch (PrivilegedActionException e)
             {
                 throw new RuntimeException(e.getException());
             }
         }
 
         @Override
-        public void onAuthenticationSuccess(byte[] token)
+        public CompletionStage<Void> onAuthenticationSuccess(@Nullable ByteBuffer var1)
```

Comment:
Missing NonNull annotation, also consider reverting the argument name var1 is not very descriptive. We may not be using it but it is good to be clear what's there.
```suggestion
        @NonNull
        public CompletionStage<Void> onAuthenticationSuccess(@Nullable ByteBuffer token)
```